kit/transport/nats/consumer.go

```diff
@@ -0,0 +1,223 @@
+	}
+}
+
+func WithOptStartTime(optStartTime *time.Time) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.OptStartTime = optStartTime
+	}
+}
+
+func WithAckPolicy(ackPolicy natn.AckPolicy) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.AckPolicy = ackPolicy
+	}
+}
+
+func WithAckWait(ackWait time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.AckWait = ackWait
+	}
+}
+
+func WithMaxDeliver(maxDeliver int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxDeliver = maxDeliver
+	}
+}
+
+func WithFilterSubject(filterSubject string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.FilterSubject = filterSubject
+	}
+}
+
+func WithReplayPolicy(replayPolicy natn.ReplayPolicy) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.ReplayPolicy = replayPolicy
+	}
+}
+
+func WithRateLimit(rateLimit uint64) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.RateLimit = rateLimit
+	}
+}
+
+func WithSampleFrequency(sampleFrequency string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.SampleFrequency = sampleFrequency
+	}
+}
+
+func WithMaxAckPending(maxAckPending int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxAckPending = maxAckPending
+	}
+}
+
+func WithFlowControl(flowControl bool) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.FlowControl = flowControl
+	}
+}
+
+func WithHeartbeat(heartbeat time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.Heartbeat = heartbeat
+	}
+}
+
+func WithMaxRequestBatch(maxRequestBatch int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxRequestBatch = maxRequestBatch
+	}
+}
+
+func WithMaxRequestExpires(maxRequestExpires time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxRequestExpires = maxRequestExpires
+	}
+}
+
+func WithInactiveThreshold(inactiveThreshold time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.InactiveThreshold = inactiveThreshold
+	}
+}
+
+func defaultConsumerErrorHandler(cx context.Context, err error) error {
+	return err
+}
+
+func NewConsumer(connstr string, stream string, options ...ConsumerOption) (*Consumer, error) {
+	var (
+		err  error
+		cc   *natn.Conn
+		opts = natn.GetDefaultOptions()
+		c    = &Consumer{
+			conn:         nil,
+			opts:         &opts,
+			cfg:          &natn.ConsumerConfig{},
+			info:         &natn.ConsumerInfo{},
+			errorHandler: defaultConsumerErrorHandler,
+		}
+	)
+
+	for _, fn := range options {
+		fn(c)
+	}
+
+	c.opts.Url = connstr
+
+	// Connect to NATS
+	cc, err = c.opts.Connect()
+
+	if err != nil {
+		return nil, errors.Wrap(
+			err, "unable to connect to nats server",
+		)
+	}
+
+	// Create JetStream Context
+	js, _ := cc.JetStream()
+
+	// Create a consumer
+	info, err := js.AddConsumer(stream, c.cfg)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create consumer")
+	}
+
+	c.js = js
+	c.conn = cc
+	c.info = info
+
+	return c, nil
+}
+
+func (c *Consumer) GetConsumerInfo() (*natn.ConsumerInfo, error) {
```

Comment:
Can you use an internal object to `go-base` as output of this method? This leaks the internal details which will then require the upstream to import nats dependency.